### PR TITLE
twister: Always change skip to error on integration platforms

### PR DESCRIPTION
**Pull request by PerMac**
Before only with --integration such skips were changed to errors. This hinders issues when twister is call with --all flag in CI and new skips on integration platforms are not caught. Then those skips cause errors in other PRs when --integration was used.

### CODE REVIEW

**Comment by PerMac** [comment]
This change prevents issues like this:
https://github.com/zephyrproject-rtos/zephyr/issues/62380

**Comment by nashif** [comment]
you need to fix the testsuite.

**Review by nashif — APPROVED** [review]

**Review by hakehuang — APPROVED** [review]

File: scripts/pylib/twister/twisterlib/testplan.py [hunk 1]
```diff
@@ -1021,8 +1021,8 @@ def _create_build_dir_link(self, links_dir_path, instance):
 
 
 def change_skip_to_error_if_integration(options, instance):
-    ''' If integration mode is on all skips on integration_platforms are treated as errors.'''
-    if options.integration and instance.platform.name in instance.testsuite.integration_platforms \
+    ''' All skips on integration_platforms are treated as errors.'''
+    if instance.platform.name in instance.testsuite.integration_platforms \
         and "quarantine" not in instance.reason.lower():
```
Comment (gchwier) [inline scripts/pylib/twister/twisterlib/testplan.py:1026]: Just a comment: There is `Filters.QUARANTINE` do there is no need to check string in the reason, better to check predefined filter.

**Review by gchwier — APPROVED** [review]

File: scripts/pylib/twister/twisterlib/testplan.py (hunk 1 above)
Comment (PerMac) [reply scripts/pylib/twister/twisterlib/testplan.py:1026]: let's do this in the follow-up PR

Comment (PerMac) [reply scripts/pylib/twister/twisterlib/testplan.py:1026]: #62810

**Pull request merged**